contracts/sophia.md

```diff
@@ -241,22 +241,47 @@ Sophia has the following types:
 | bool       | A Boolean                       | ```true```
 | bits       | A bit field (with 256 bits)     | ```Bits.none```
+| bytes(n)   | A byte array with `n` bytes     | ```#fedcba9876543210```
 | string     | An array of bytes               | ```"Foo"```
 | list       | A homogeneous immutable singly linked list. | ```[1, 2, 3]```
 | tuple      | An ordered heterogeneous array   | ```(42, "Foo", true)```
 | record     | An immutable key value store with fixed key names and typed values | ``` record balance = { owner: address, value: int } ```
 | map        | An immutable key value store with dynamic mapping of keys of one type to values of one type | ```type accounts = map(string, address)```
 | option('a) | An optional value either None or Some('a) | ```Some(42)```
-| state        | A record of blockstate key, value pairs  |
-| transactions | An append only list of blockchain transactions |
-| events       | An append only list of blockchain events (or log entries) |
-| signature    | A signature - 64 bytes |
-| Chain.ttl    | Time-to-live (fixed height or relative to current block) | ```FixedTTL(1050)``` ```RelativeTTL(50)```
+| state      | A user defined type holding the contract state | ```record state = { owner: address, magic_key: bytes(4) }```
+| event      | An append only list of blockchain events (or log entries) | ```datatype event = EventX(indexed int, string)```
+| hash       | A 32-byte hash - equivalent to `bytes(32)` |
+| signature  | A signature - equivalent to `bytes(64)` |
+| Chain.ttl  | Time-to-live (fixed height or relative to current block) | ```FixedTTL(1050)``` ```RelativeTTL(50)```
 | oracle('a, 'b)       | And oracle answering questions of type 'a with answers of type 'b |  ```Oracle.register(acct, qfee, ttl)```
 | oracle_query('a, 'b) | A specific oracle query |  ```Oracle.query(o, q, qfee, qttl, rttl)```
+| contract   | A user defined, typed, contract address | ```function call_remote(r : RemoteContract) = r.fun()```
+
+### Literals
+| Type       | Constant/Literal example(s)
+| ---------- | -------------------------------
+| int        | `-1`, `2425`, `4598275923475723498573485768`
+| address    | `ak_2gx9MEFxKvY9vMG5YnqnXWv1hCsX7rgnfvBLJS4aQurustR1rt`
+| bool       | `true`, `false`
+| bits       | `Bits.none`, `Bits.all`
+| bytes(n)   | `#fedcba9876543210`
+| string     | `"This is a string"`
+| list       | `[1, 2, 3]`, `[(true, 24), (false, 19), (false, -42)]`
+| tuple      | `(42, "Foo", true)`
+| record     | `balance{ owner = Call.origin, value = 100000000 }`
+| map        | `{["foo"] = 19, ["bar"] => 42}`, `{}`
+| option('a) | `Some(42)`, `None`
+| state      | `state{ owner = Call.origin, magic_key = #a298105f }`
+| event      | `EventX(0, "Hello")`
+| hash       | `#000102030405060708090a0b0c0d0e0f000102030405060708090a0b0c0d0e0f`
+| signature  | `#000102030405060708090a0b0c0d0e0f000102030405060708090a0b0c0d0e0f000102030405060708090a0b0c0d0e0f000102030405060708090a0b0c0d0e0f`
```

Comment:
shouldn't signature be either a `byte(X)` or a special literal to use the nodes encoding using `sg_`?

Reply:
I am not sure... Since a signature is really interchangeable with `bytes(64)` it feels natural to use the same kind of literal. Also signatures might come from elsewhere (Bitcoin?) and then it does not feel natural to use Aeternity encoding :thinking:

Reply:
I do agree, I was just throwing this thought out here. But we could destinguish aeternity signatures from others.